aws-lc-rs/src/digest/sha.rs

```diff
@@ -143,3 +193,17 @@ fn sha512_256_digest(msg: &[u8], output: &mut [u8]) {
         aws_lc::SHA512_256(msg.as_ptr(), msg.len(), output.as_mut_ptr());
     }
 }
+
+fn sha3_384_digest(msg: &[u8], output: &mut [u8]) {
+    let mut ctx = Context::new(&SHA3_384);
+    ctx.update(msg);
+    let digest = ctx.finish();
+    output[0..SHA3_384_OUTPUT_LEN].copy_from_slice(digest.as_ref());
```

Comment:
I assume:
```
Context::new(&SHA3_384) calls EVP_DigestInit,
ctx.update(msg) calls EVP_DigestUpdate,
digest = ctx.finish() calls EVP_DigestFinal.
```
right?

Reply:
Right. To be specific:
* `Context::new(&SHA3_384)` calls [`EVP_MD_CTX_init` and `EVP_DigestInit_ex`](https://github.com/awslabs/aws-lc-rs/blob/main/aws-lc-rs/src/digest/digest_ctx.rs#L17-L18).
* `ctx.update(msg)` calls [`EVP_DigestUpdate`](https://github.com/awslabs/aws-lc-rs/blob/main/aws-lc-rs/src/digest.rs#L105)
* `ctx.finish()` calls [`EVP_DigestFinal`](https://github.com/awslabs/aws-lc-rs/blob/main/aws-lc-rs/src/digest.rs#L131)
* When the value goes out of scope, [`EVP_MD_CTX_cleanup`](https://github.com/awslabs/aws-lc-rs/blob/main/aws-lc-rs/src/digest/digest_ctx.rs#L46) is called.